### PR TITLE
2015 09 mn create metrics

### PR DESCRIPTION
**Pull request by MagdaN**
this is not finished yet.

### CODE REVIEW

**Comment by slomo** [comment]
I will try push this forward, while @MagdaN is out of office

**Comment by FabiApfelkern** [comment]
I will review this.

**Comment by FabiApfelkern** [comment]
Can I get an overview, what is suppose to work with this PR?

**Comment by slomo** [comment]
This is not ready for review.

**Comment by FabiApfelkern** [comment]
Ok. I thought so, because Nico said yesterday we will see something about MM frontend. But nevermind,

**Comment by MagdaN** [comment]
@slomo : can you review and merge

File: app/modules/metrics/controllers/metricControllers.js [hunk 1]
```diff
@@ -8,436 +8,244 @@ angular.module('pcApp.metrics.controllers.metric', [
...
-			},
-			function(error) {
-                throw { message: JSON.stringify(err.data)};
-			}
-	);
+    $scope.getCursorPosition = function(event) {
+        var element = event.target;
+        $scope.doGetCaretPosition(element);
+
+        /*var caretOffset = 0;
```
Comment (slomo) [inline app/modules/metrics/controllers/metricControllers.js]: Lots of commented code lying around here.

File: app/modules/metrics/controllers/metricControllers.js [hunk 2]
```diff
@@ -8,436 +8,244 @@ angular.module('pcApp.metrics.controllers.metric', [
...
+
+    $scope.addIndicator = function(indicator) {
+
+        var i = "__" + $scope.variableIndex + "__";
+        if (angular.isUndefined($scope.cursorPosVal) || angular.isUndefined($scope.metrics_controller_helper.metricsdata.formula)){
+            if(angular.isUndefined($scope.metrics_controller_helper.metricsdata.formula)){
+                $scope.metrics_controller_helper.metricsdata.formula = "";
+            }
+            $scope.metrics_controller_helper.metricsdata.formula = $scope.metrics_controller_helper.metricsdata.formula + i;
+            //$("#formula-div").append('<span contentEditable="false" class="indicator-formula indicator-formula-selected">' + indicator.name + '</span>');
```
Comment (slomo) [inline app/modules/metrics/controllers/metricControllers.js]: More commented code here.

**Pull request merged**